### PR TITLE
first task in ror, about todo list

### PR DESCRIPTION
**Pull request by Serghei88**

### CODE REVIEW

**Comment by Serghei88** [comment]
@vlebedeff could you please take a look?

File: tasks/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (vlebedeff) [inline tasks/.vscode/launch.json:1]: This file probably shouldn't be in the Git repo

File: tasks/app/views/tasks/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <th></th>
+  </tr>
+
+<% @tasks.each do |task| %>
+  <tr>
+    <td><%= task["name"] %></td>
+    <td><%= task["description"] %></td>
+    <td><%= link_to 'Show', task_path(task["id"]) %></td>
+    <td><%= link_to 'Edit', edit_task_path(task["id"]) %></td>
+    <td><%= link_to 'Destroy', task_path(task["id"]), method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (vlebedeff) [inline tasks/app/views/tasks/index.html.erb]: You should be able to get task attributes using normal methods `task.id / task.description / task.name`
Any reason for using Hash-like syntax?

File: tasks/app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+class TasksController < ApplicationController
+  # GET /tasks
+  # GET /tasks.json
+  def index
+    tasks = $redis.get('tasks')
+
+    if tasks.nil?
+      tasks = Task.all.to_json
+      $redis.set('tasks', tasks)
+      $redis.expire('tasks', 2.minute.to_i)
```
Comment (vlebedeff) [inline tasks/app/controllers/tasks_controller.rb]: Too much responsibility for a controller.
I think you could add data access layer for tasks.
Some `TaskRepository` class that would know how to get tasks from database/redis and save them.

**Comment by Serghei88** [comment]
done

File: tasks/app/models/task.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Task < ActiveRecord::Base
+  attr_accessible :description, :name
+
+  after_save :clear_cache
+  after_update :clear_cache
+  after_destroy :clear_cache
+
+
+  def clear_cache
+    $redis.del "tasks"
```
Comment (vlebedeff) [inline tasks/app/models/task.rb]: I think, this logic is better now to be in that TaskRepository you have

File: tasks/app/views/tasks/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <th></th>
+    <th></th>
+    <th></th>
+  </tr>
+
+<% @tasks.each do |task| %>
+  <tr>
+    <td><%= task.name %></td>
+    <td><%= task.description %></td>
+    <td><%= link_to 'Show', task_path(task["id"]) %></td>
```
Comment (vlebedeff) [inline tasks/app/views/tasks/index.html.erb]: `task["id"] -> task.id`

File: tasks/app/views/tasks/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <td><%= link_to 'Edit', edit_task_path(task["id"]) %></td>
```
Comment (vlebedeff) [inline tasks/app/views/tasks/index.html.erb]: `task["id"] -> task.id`

File: tasks/app/views/tasks/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    <td><%= link_to 'Destroy', task_path(task["id"]), method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (vlebedeff) [inline tasks/app/views/tasks/index.html.erb]: `task["id"] -> task.id`

**Comment by Serghei88** [comment]
done with previous comments

**Review by vlebedeff — APPROVED** [review]

**Pull request merged**